pinot-query-planner/src/test/resources/queries/SetOpPlans.json

```diff
@@ -0,0 +1,95 @@
+{
+  "set_op_tests": {
+    "queries": [
+      {
+        "description": "UNION ALL from two tables",
+        "sql": "EXPLAIN PLAN FOR SELECT col1, col2 FROM a UNION ALL SELECT col1, col2 FROM b",
+        "output": [
+          "Execution Plan",
+          "\nLogicalUnion(all=[true])",
+          "\n  LogicalProject(col1=[$0], col2=[$1])",
+          "\n    LogicalTableScan(table=[[a]])",
+          "\n  LogicalProject(col1=[$0], col2=[$1])",
+          "\n    LogicalTableScan(table=[[b]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "UNION ALL from three tables",
+        "sql": "EXPLAIN PLAN FOR SELECT col1, col2 FROM a UNION ALL SELECT col1, col2 FROM b UNION ALL SELECT col1, col2 FROM c",
+        "output": [
+          "Execution Plan",
+          "\nLogicalUnion(all=[true])",
+          "\n  LogicalUnion(all=[true])",
+          "\n    LogicalProject(col1=[$0], col2=[$1])",
+          "\n      LogicalTableScan(table=[[a]])",
+          "\n    LogicalProject(col1=[$0], col2=[$1])",
+          "\n      LogicalTableScan(table=[[b]])",
+          "\n  LogicalProject(col1=[$0], col2=[$1])",
+          "\n    LogicalTableScan(table=[[c]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "UNION from three tables",
+        "sql": "EXPLAIN PLAN FOR SELECT col1, col2 FROM a UNION SELECT col1, col2 FROM b UNION SELECT col1, col2 FROM c",
+        "output": [
+          "Execution Plan",
+          "\nLogicalAggregate(group=[{0, 1}])",
+          "\n  LogicalExchange(distribution=[hash[0, 1]])",
+          "\n    LogicalAggregate(group=[{0, 1}])",
+          "\n      LogicalUnion(all=[true])",
+          "\n        LogicalUnion(all=[true])",
+          "\n          LogicalProject(col1=[$0], col2=[$1])",
+          "\n            LogicalTableScan(table=[[a]])",
+          "\n          LogicalProject(col1=[$0], col2=[$1])",
+          "\n            LogicalTableScan(table=[[b]])",
+          "\n        LogicalProject(col1=[$0], col2=[$1])",
+          "\n          LogicalTableScan(table=[[c]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "INTERSECT from three tables",
+        "sql": "EXPLAIN PLAN FOR SELECT col1, col2 FROM a INTERSECT SELECT col1, col2 FROM b INTERSECT SELECT col1, col2 FROM c",
```

Comment:
@xiangfu0 : Shouldn't intersect require an exchange?

If the row `col1=10, col2=11` in tables `a` and `b` are on different servers, if we don't shuffle using the intersection key (or broadcast one side) we may end up dropping them from the result-set.

Reply:
That was planned to be added in the next PR:

https://github.com/apache/pinot/pull/10622/files#diff-1216478b200657de2afb3b0c8552bda799e67b68049aa025dbefc037235d9fb0